src/Controls/src/Core/HandlerImpl/SwipeView/SwipeView.Impl.cs

```diff
@@ -112,6 +116,51 @@ private protected override void OnParentChangedCore()
 			}
 		}
 
+		internal void AddLogicalChild(Element element)
+		{
+			if (element is null)
+			{
+				return;
+			}
+
+			if (_logicalChildren.Contains(element))
+				return;
+
+			_logicalChildren.Add(element);
+			element.Parent = this;
+			OnChildAdded(element);
+			VisualDiagnostics.OnChildAdded(this, element);
+		}
+
+		internal void ClearLogicalChildren()
+		{
+			// Reverse for-loop, so children can be removed while iterating
+			for (int i = _logicalChildren.Count - 1; i >= 0; i--)
+			{
+				RemoveLogicalChildByIndex(i);
+			}
+		}
+
+		void RemoveLogicalChildByIndex(int index)
+		{
+			if (_logicalChildren.Count < index)
+			{
+				return;
+			}
+
+			var element = _logicalChildren[index];
+
+			if (element is null)
+			{
+				return;
+			}
+
+			element.Parent = null;
+			_logicalChildren.Remove(element);
```

Comment:
Can we use https://learn.microsoft.com/en-us/dotnet/api/system.collections.objectmodel.observablecollection-1.removeitem?view=net-7.0 to remove the element in the given index. Maybe wrap around a try if you are not sure of the index.

Reply:
We had an off github chat about this, @jsuarezruiz correctly mentioned that he needs the element to use it. Getting the element is a O(1) so we are not worried about that. The question was, once we had the element, which is better, Remove or RemoveItem? 

Looking at the reference implementation of Collection<T> which is the one that provides Remove, we can see that it does another lookup to get the index, so using RemoveItem is more efficient.

Reference: https://github.com/microsoft/referencesource/blob/master/mscorlib/system/collections/objectmodel/collection.cs#L106

Reply:
Yeah, nice conversations. Thanks for the feedback.